src/adaptation/icap/ModXact.cc

```diff
@@ -1167,6 +1192,8 @@ void Adaptation::Icap::ModXact::parseBody()
     // the parser will throw on errors
     BodyPipeCheckout bpc(*adapted.body_pipe);
     bodyParser->setPayloadBuffer(&bpc.buf);
+    ExtensionsParser extensionsParser;
+    bodyParser->setCustomExtensionsParser(&extensionsParser);
```

Comment:
It feels wrong to (re)set the extensions parser every time we parse a piece of response body. I suggest doing it once, when we create the parser. This will require making extensionsParser a class member, but that is OK.

Making the above change will also address extensionsParser lifetime concern: This may not be possible in the current code, but, technically, since bodyParser is an incremental parser, bodyParser->parse() may return false _after_ successfully parsing and consuming the use-original-body extension value. In that case, that parsed value will be lost in the current code (because we will reset extensionsParser on the next parsing path).